### PR TITLE
feat(pagination): add change detection to angular pagination

### PR DESCRIPTION
**Pull request by adsmith2**
add changing the component view in response to changes in inputs for current page, total pages, and group sizes

The input properties for pagination will now dynamically change the view. New tests were added to test dynamic changes.

## Screenshots:

**After**:

Dynamic_Pagination

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: angular/src/lib/pagination/test/pagination.component.spec.ts [hunk 1]
```diff
@@ -3,106 +3,209 @@ import { PaginationService } from '../pagination.service';
...
+      component.current = 5;
+      component.ngOnInit();
+      fixture.detectChanges();
+
+      const ellipsis: DebugElement[] = fixture.debugElement.queryAll(By.css('li.pagination_li.ellipsis'));
+      expect(ellipsis.length).toBe(0);
+
+      const numberComponents: DebugElement[] = fixture.debugElement.queryAll(By.directive(PaginationNumberComponent));
+      const numberComponentsText = numberComponents
+        .reduce((accumulator, currentElement) => accumulator + ' ' + currentElement.nativeElement.textContent, '');
```
[the remark this answers is not in the document]
Comment (adsmith2) [reply angular/src/lib/pagination/test/pagination.component.spec.ts]: Very useful. Thanks!

File: angular/src/lib/pagination/pagination-arrow.component.ts [hunk 2]
```diff
@@ -46,6 +46,9 @@ export class PaginationArrowComponent implements OnInit {
     this.paginationService.current$.subscribe(() => {
       this.build();
     });
+    this.paginationService.total$.subscribe(() => {
```
[the remark this answers is not in the document]
Comment (adsmith2) [reply angular/src/lib/pagination/pagination-arrow.component.ts]: I fixed this, and fixed several other existing subscriptions in the same component that had the same problem. Can you review these changes as well?

File: angular/src/lib/pagination/test/pagination.component.spec.ts [hunk 3]
```diff
@@ -3,106 +3,205 @@ import { PaginationService } from '../pagination.service';
...
+      expect(paginationService).toBeTruthy();
+    });
+
+    it('should render total 3', () => {
+      component.total = 3;
+      component.current = 2;
+      component.ngOnInit();
+      fixture.detectChanges();
+      testElements = fixture.nativeElement.getElementsByTagName('a');
+      expect(testElements.length).toEqual(4);
```
Comment (mathsusan) [inline angular/src/lib/pagination/test/pagination.component.spec.ts]: why are there 4 when we are looking for 3?  Do this test really test what we think it does?

File: angular/src/lib/pagination/test/pagination.component.spec.ts [hunk 4]
```diff
@@ -3,106 +3,205 @@ import { PaginationService } from '../pagination.service';
...
+    });
+
+    it('should disable previous', () => {
+      component.total = 3;
+      component.ngOnInit();
+      fixture.detectChanges();
+      testElement = fixture.nativeElement.getElementsByTagName('md-pagination-arrow')[0];
+      expect(testElement.className).toContain('icon-arrow-left-optical_20');
```
Comment (mathsusan) [inline angular/src/lib/pagination/test/pagination.component.spec.ts]: how does this test that the previous button is disabled?  It is only looking for the left arrow which I expect would be there either way

File: angular/src/lib/pagination/test/pagination.component.spec.ts [hunk 5]
```diff
@@ -3,106 +3,205 @@ import { PaginationService } from '../pagination.service';
...
+      fixture.detectChanges();
+
+      testElement = fixture.nativeElement.getElementsByTagName('md-pagination-arrow')[1];
+      expect(testElement.className).not.toContain('icon-arrow-right-optical_20');
+
+      component.totalPages = 5;
+      fixture.detectChanges();
+
+      expect(testElement.className).toContain('icon-arrow-right-optical_20');
+    });
```
Comment (mathsusan) [inline angular/src/lib/pagination/test/pagination.component.spec.ts]: same comment as on previous disable.  How does this test that it is disabled?

File: angular/src/lib/pagination/test/pagination.component.spec.ts (hunk 3 above)
Comment (adsmith2) [reply angular/src/lib/pagination/test/pagination.component.spec.ts]: I think the test is checking that there are 4 clickable elements (< 1  3 > ), but yeah, the test description definitely doesn't reflect that. I'll update that.

**Pull request merged**